terminal/terminal.go

```diff
@@ -150,6 +167,34 @@ func (t *Term) Println(prefix, str string) {
 	fmt.Fprintf(t.stdout, "%s%s\n", prefix, str)
 }
 
+// SubstitutePath substitues directory to soruce file.
+//
+// Ensures that only directory is substitued, for example:
+// substitute from `/dir/subdir`, substitute to `/new`
+// for file path `/dir/subdir/file` will return file path `/new/file`.
+// for file path `/dir/subdir-2/file` substitution will not be applied.
+//
+// If more than one substitution rule is defiend, the rules are applied
+// in the order they are defined, first rule that matches is used for
+// substitution.
+func (t *Term) SubstitutePath(path string) string {
+	separator := string(os.PathSeparator)
```

Comment:
> case sensitivity depends on file system

I have never used Windows OS that cared about file path case. I do not know what else I can say.

> Do you have some hints on that?

I would always assume that file path is case insensitive on Windows.

> I'm not sure if it's ok to do it based on the type of operating system (but this would be consistent with gdb).

Converting to lower case on Windows before comparing strings sounds good to me.

There is also path/filepath.EvalSymlinks which will convert your file path to the same form (regardless case and everything else). But filepath.EvalSymlinks works with real files only - the file must exist.

Alex